File: Roadmap/06 - RECURSIVIDAD/javascript/garos01.js
function imprimirNumerosDecrecientes(numero) {
  // Caso base
  if (numero < 0) {
    return;
  }

  console.log(numero);

  // Llamada recursiva
  imprimirNumerosDecrecientes(numero - 1);
}

// Uso
imprimirNumerosDecrecientes(100);

// Ejercicio extra
// Factorial
function calcularFactorial(numero) {
  if (numero === 0) {
    return 1;
  } else {
    return numero * calcularFactorial(numero - 1);
  }
}

const numero = 5;
const factorial = calcularFactorial(numero);

console.log(`El factorial de ${numero} es: ${factorial}`);

// Posicion Fibonacci
function fibonacci(posicion) {
  if (posicion === 1 || posicion === 2) {
    return 1;
  } else {
    return fibonacci(posicion - 1) + fibonacci(posicion - 2);
  }
}

const posicion = 10;
const valorEnPosicion = fibonacci(posicion);

console.log(
  `El valor en la posición ${posicion} de la sucesión de Fibonacci es: ${valorEnPosicion}`
);

File: Roadmap/06 - RECURSIVIDAD/python/garos01.py
# Recursividad


def decremento(numero):
    if numero < 0:
        return
    print(numero)
    decremento(numero - 1)


decremento(100)


# Ejercicio extra


# Factorial
def factorial(numero):
    if numero == 0:
        return 1
    else:
        return numero * factorial(numero - 1)


print(f"Factorial {factorial(5)}")


# Fibonacci
def fibonacci(posicion):
    if posicion == 1 or posicion == 2:
        return 1
    else:
        return fibonacci(posicion - 1) + fibonacci(posicion - 2)


print(f"Numero {fibonacci(10)}")
